### PR TITLE
Remove duplication on the Connector Autofac modules

### PR DESCRIPTION
**Pull request by adamhathcock**
CNX-9433

First pass: rename to SpeckleContainerBuilder to separate concerns

### CODE REVIEW

**Comment by adamhathcock** [comment]
Made:
SpeckleContainer
SpeckleContainerBuilder
ISpeckleModule

Hides Autofac now and will load things from ISpeckleModule

**Comment by adamhathcock** [comment]
every host app has two ISpeckleModule: one for Converters and one for Connectors.  

Connectors should have DUI and stuff
Converters are just converter code.

This rule isn't clean because things still reference each other

**Review by AlanRynne — COMMENTED** [review]
I realised i did my comments but never sent the review 🙇🏼‍♂️

This is so much cleaner ♥️ I've got some comments/questions but I see no blockers for this to go in.

File: DUI3-DX/Connectors/ArcGIS/Speckle.Connectors.ArcGIS3/DependencyInjection/ArcGISConnectorModule.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    builder.AddSingleton<IBinding, TestBinding>();
+    builder.AddSingleton<IBinding, ConfigBinding>("connectorName", "ArcGIS"); // POC: Easier like this for now, should be cleaned up later
+    builder.AddSingleton<IBinding, AccountBinding>();
+    builder.AddSingleton<IBinding, BasicConnectorBinding>();
+    builder.AddSingleton<IBasicConnectorBinding, BasicConnectorBinding>();
+    builder.AddSingleton<IBinding, ArcGISSelectionBinding>();
+    builder.AddSingleton<IBinding, ArcGISSendBinding>();
+    builder.AddSingleton<IBinding, ArcGISReceiveBinding>();
+
+    builder.AddTransient<ISendFilter, ArcGISSelectionFilter>();
```
[the remark this answers is not in the document]
Comment (adamhathcock) [reply DUI3-DX/Connectors/ArcGIS/Speckle.Connectors.ArcGIS3/DependencyInjection/ArcGISConnectorModule.cs:41]: I'm trying to copy one to one and ask questions.   I think most things should be transient and not scoped.  I need to better understand though

File: DUI3-DX/Connectors/Autocad/Speckle.Connectors.AutocadShared/Plugin/AutocadCommand.cs [hunk 2]
```diff
@@ -36,12 +33,12 @@ public void Command()
       DockEnabled = (DockSides)((int)DockSides.Left + (int)DockSides.Right)
     };
 
-    Container = new AutofacContainer(new StorageInfo());
-    Container.PreBuildEvent += ContainerPreBuildEvent;
+    var builder = SpeckleContainerBuilder.CreateInstance();
+    builder.AddConverterCommon();
```
[the remark this answers is not in the document]
Comment (adamhathcock) [reply DUI3-DX/Connectors/Autocad/Speckle.Connectors.AutocadShared/Plugin/AutocadCommand.cs:37]: definitely don't need that event.  it was an artifact of having combined a builder with the container.  I've split them again

File: DUI3-DX/Sdk/Speckle.Converters.Common.DependencyInjection/ContainerRegistration.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    speckleContainerBuilder.InjectNamedTypes<IHostObjectToSpeckleConversion>();
+    speckleContainerBuilder.InjectNamedTypes<ISpeckleObjectToHostConversion>();
+
+    // POC: will likely need refactoring with our reporting pattern.
+    var serilogLogger = new LoggerConfiguration().MinimumLevel
+      .Debug()
+      .WriteTo.File("log.txt", rollingInterval: RollingInterval.Day)
+      .CreateLogger();
+
+    ILoggerFactory loggerFactory = new LoggerFactory().AddSerilog(serilogLogger);
```
[the remark this answers is not in the document]
Comment (adamhathcock) [reply DUI3-DX/Sdk/Speckle.Converters.Common.DependencyInjection/ContainerRegistration.cs]: Moved.  Makes sense to be part of connectors

File: DUI3-DX/Sdk/Speckle.Autofac/EnumerationExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+public static class EnumerationExtensions
+{
+  public static void ForEach<T>(this IEnumerable<T> items, Action<T> action)
+  {
+    foreach (var item in items)
+    {
+      action(item);
+    }
+  }
```
[the remark this answers is not in the document]
Comment (adamhathcock) [reply DUI3-DX/Sdk/Speckle.Autofac/EnumerationExtensions.cs:11]: for anything that can be enumeration and need an action on it with no return

File: DUI3-DX/Sdk/Speckle.Converters.Common.DependencyInjection/NamedTypeInjector.cs [hunk 5]
```diff
@@ -1,32 +1,16 @@
...
+using Speckle.Autofac.DependencyInjection;
 using Speckle.Converters.Common.Objects;
-using Speckle.Core.Logging;
 
 namespace Speckle.Converters.Common.DependencyInjection;
 
 public static class ConversionTypesInjector
 {
-  public static ContainerBuilder InjectNamedTypes<T>(this ContainerBuilder containerBuilder)
+  public static void InjectNamedTypes<T>(this SpeckleContainerBuilder containerBuilder)
```
[the remark this answers is not in the document]
Comment (adamhathcock) [reply DUI3-DX/Sdk/Speckle.Converters.Common.DependencyInjection/NamedTypeInjector.cs:10]: Can do that but no one is.  Anyone can change it.

File: DUI3-DX/Sdk/Speckle.Converters.Common.DependencyInjection/NamedTypeInjector.cs [hunk 6]
```diff
@@ -49,7 +33,7 @@ public static ContainerBuilder InjectNamedTypes<T>(this ContainerBuilder contain
       var first = namedTypes[0];
 
       // POC: may need to be instance per lifecycle scope
-      containerBuilder.RegisterType(first.type).Keyed<T>(first.name).InstancePerLifetimeScope();
+      containerBuilder.ContainerBuilder.RegisterType(first.type).Keyed<T>(first.name).InstancePerLifetimeScope();
```
[the remark this answers is not in the document]
Comment (adamhathcock) [reply DUI3-DX/Sdk/Speckle.Converters.Common.DependencyInjection/NamedTypeInjector.cs:36]: Not final :)

**Comment by adamhathcock** [comment]
Still gotta test and see if initialization is happening.

**Pull request closed without merging**